capstone/backend/src/main/java/com/google/servlets/CreateUserServlet.java

```diff
@@ -74,9 +75,11 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     try {
       createUser(userId, (String) payload.get("name"), payload.getEmail());
     } catch (SpannerException se) {
-      response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR,
+      if (se.getErrorCode() != ErrorCode.ALREADY_EXISTS) {
```

Comment:
Actually an easy fix would be to change the library function to `newInsertOrUpdateBuilder` so this won't fail